app/screens/permalink/permalink.js

```diff
@@ -0,0 +1,458 @@
+            const currentChannelId = this.props.channelId;
+            const {
+                handleSelectChannel,
+                handleTeamChange,
+                markChannelAsRead,
+                setChannelLoading,
+                setChannelDisplayName,
+                markChannelAsViewed
+            } = actions;
+
+            if (this.refs.view) {
+                this.refs.view.zoomOut().then(() => {
+                    actions.selectPost('');
+
+                    if (onClose) {
+                        onClose();
+                    }
+
+                    navigator.resetTo({
+                        screen: 'Channel',
+                        animated: true,
+                        animationType: 'fade',
+                        navigatorStyle: {
+                            navBarHidden: true,
+                            statusBarHidden: false,
+                            statusBarHideWithNavBar: false,
+                            screenBackgroundColor: theme.centerChannelBg
+                        }
+                    });
+
+                    if (channelTeamId && currentTeamId !== channelTeamId) {
+                        handleTeamChange(channelTeamId, false);
+                    }
+
+                    setChannelLoading(channelId !== currentChannelId);
+                    setChannelDisplayName(channelDisplayName);
+                    handleSelectChannel(channelId);
+
+                    InteractionManager.runAfterInteractions(async () => {
+                        markChannelAsRead(channelId, currentChannelId);
+                        if (channelId !== currentChannelId) {
+                            markChannelAsViewed(currentChannelId);
+                        }
+                    });
+                });
+            }
+        }
+    };
+
+    handlePress = () => {
+        const {channelId, channelName} = this.props;
+
+        if (this.refs.view) {
+            this.refs.view.growOut().then(() => {
+                this.jumpToChannel(channelId, channelName);
+            });
+        }
+    };
+
+    loadPosts = async (props, context) => {
+        const {intl} = this.context || context;
+        const {actions, channelId, currentUserId, focusedPostId, isPermalink, postIds} = props;
+        const {formatMessage} = intl;
+        let focusChannelId = channelId;
+
+        const post = await actions.getPostThread(focusedPostId, false);
+        if (post.error && (!postIds || !postIds.length)) {
+            if (isPermalink && post.error.message.toLowerCase() !== 'network request failed') {
+                this.setState({
+                    error: formatMessage({
+                        id: 'permalink.error.access',
+                        defaultMessage: 'Permalink belongs to a deleted message or to a channel to which you do not have access.'
+                    }),
+                    title: formatMessage({
```

Comment:
Instead of needing to track the title separately, couldn't we just have the render function show this when there's an error or just show the channel's name otherwise?

Reply:
if this happens you may not have the channel name, that is why we are changing the title here